### PR TITLE
[Intl] Generate all emoji short name returned by slack api

### PR DESCRIPTION
**Pull request by adnen-chouibi**
| Q             | A
| ------------- | ---
| Branch?       | 6.2 
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       |  no
| License       | MIT
| Doc PR        |  NA

I have been utilizing the slack-emoji helper to convert country codes to flag emojis. I propose to standardize all flags by using the format :flag-countryCode: for all flags, or maybe create another helper especially for flags. Additionally, I suggest incorporating all emojis returned in the 'short_names' array, rather than relying solely on the 'short_name' value. I have improved the file naming process by generating the file name in the builder, rather than in the 'saveRules' method.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Thanks, can you update the generated files in the same PR please?

File: src/Symfony/Component/Intl/Resources/emoji/build.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ public static function buildGitHubRules(array $emojisCodePoints): iterable
             $maps[$codePointsCount][$emoji] = ":$shortCode:";
         }
 
-        return ['github' => self::createRules($maps)];
+        return ['emoji-github' => self::createRules($maps), 'github-emoji' => array_flip(self::createRules($maps))];
```
Comment (stof) [inline src/Symfony/Component/Intl/Resources/emoji/build.php]: why creating rules twice here ?

File: src/Symfony/Component/Intl/Resources/emoji/build.php [hunk 2]
```diff
@@ -143,20 +143,29 @@ public static function buildSlackRules(array $emojisCodePoints): iterable
         foreach ($emojis as $data) {
             $emojiCodePoints = str_replace('-', ' ', strtolower($data['unified']));
             $shortCode = $data['short_name'];
+            $shortCodes = $data['short_names'];
+            array_walk($shortCodes, function (&$value, $key) {
```
Comment (stof) [inline src/Symfony/Component/Intl/Resources/emoji/build.php]: I'd rather use `array_map` than `array_walk` that changes the array by reference. It makes the code easier to read.

File: src/Symfony/Component/Intl/Resources/emoji/build.php [hunk 3]
```diff
@@ -143,20 +143,29 @@ public static function buildSlackRules(array $emojisCodePoints): iterable
...
                 ];
                 continue;
             }
             $emoji = $emojisCodePoints[$emojiCodePoints];
             self::testEmoji($emoji, 'slack');
             $codePointsCount = mb_strlen($emoji);
-            $maps[$codePointsCount][$emoji] = ":$shortCode:";
+            $maps['emoji-slack'][$codePointsCount][$emoji] = ":$shortCode:";
+            foreach ($shortCodes as $short_name) {
+                $maps['slack-emoji'][$codePointsCount][$short_name] = $emoji;
```
Comment (stof) [inline src/Symfony/Component/Intl/Resources/emoji/build.php]: I'd rather create 2 variables than having `$maps['emoji-slack']` and `$maps['slack-emoji']` as the actual maps

Comment (stof) [inline src/Symfony/Component/Intl/Resources/emoji/build.php]: I'd rather create 2 variables than having `$maps['emoji-slack']` and `$maps['slack-emoji']` as the actual maps

File: src/Symfony/Component/Intl/Resources/emoji/build.php (hunk 1 above)
Comment (adnen-chouibi) [reply src/Symfony/Component/Intl/Resources/emoji/build.php]: The existing code has already produced two files, "emoji-github.php" and "github-emoji", to facilitate conversion from emoji to Github code and vice versa

The deleted lines have been replaced with the new code:  

https://github.com/symfony/symfony/pull/49147/commits/ab7d00967b571eebf46fb198c540230602504147#diff-64df07e7ad990f3a461c8f1f4af4b7496fe2df6feb757214e78f27ee72db8eebL172

https://github.com/symfony/symfony/pull/49147/commits/ab7d00967b571eebf46fb198c540230602504147#diff-64df07e7ad990f3a461c8f1f4af4b7496fe2df6feb757214e78f27ee72db8eebL184

**Comment by adnen-chouibi** [comment]
I suggest building a new file flag-emoji.php, to convert country codes to flags. This new feature could enhance the Intl component by adding a convenient method, such as Countries::getFlag('GB'), which would return 🇬🇧.

File: src/Symfony/Component/Intl/Resources/emoji/build.php (hunk 1 above)
Comment (stof) [reply src/Symfony/Component/Intl/Resources/emoji/build.php]: Instead of calling `createRules` twice, you could create it once and reuse it to flip it.

**Comment by stof** [comment]
@adnen-chouibi that feature does not depend on building a map. See https://github.com/thekelvinliu/country-code-emoji for the JS implementation of such a feature, which relies only on the knowledge of how emoji flags are encoded.
This is not even a work for the symfony/intl component IMO.

**Comment by nicolas-grekas** [comment]
Thank you @adnen-chouibi.

**Pull request merged**